apps/src/blockly/googleBlocklyWrapper.js

```diff
@@ -661,24 +667,48 @@ function initializeBlocklyWrapper(blocklyInstance) {
     const trashcan = new CdoTrashcan(workspace);
     trashcan.init();
 
+    // Hidden workspace where we can put function definitions.
+    const procedureWorkspace = new Blockly.Workspace();
+    blocklyWrapper.setProcedureWorkspace(procedureWorkspace);
+
     if (options.useModalFunctionEditor) {
-      // Customize auto-populated Functions toolbox category.
-      workspace.registerToolboxCategoryCallback(
-        'PROCEDURE',
-        functionsFlyoutCategory
+      blocklyWrapper.functionEditor = new FunctionEditor();
+      // TODO: Is this the best place to call the init function
+      // (and is using the init function) reasonable for the modal function editor?
+      blocklyWrapper.functionEditor.init(
+        workspace,
+        procedureWorkspace,
+        opt_options
       );
     }
+    blocklyWrapper.setMainWorkspace(workspace);
 
     return workspace;
   };
 
   // Used by StudioApp to tell Blockly to resize for Mobile Safari.
   blocklyWrapper.fireUiEvent = function (element, eventName, opt_properties) {
     if (eventName === 'resize') {
-      blocklyWrapper.svgResize(blocklyWrapper.getMainWorkspace());
+      blocklyWrapper.svgResize(blocklyWrapper.mainBlockSpace);
     }
   };
 
+  blocklyWrapper.setMainWorkspace = function (mainWorkspace) {
```

Comment:
So we have two copies of this function, right? One in the googleBlocklyWrapper and one in the functionEditor? Do you foresee any issues potentially with being able to set one without the other/keeping them in sync?

Reply:
It looks like the way you call them, they shouldn't get out of sync, but maybe it's worth noting the distinction?

Reply:
Good question. We really don't need the one in the function editor, we can just rely on the one in the blockly wrapper (they are always the same). I'll update to remove the one from the function editor.